### PR TITLE
rec RPZ dumpFile/seedFile: store/get SOA refresh on dump/load

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
Without this, we would store the refresh value as zero, and ignore it on load (also making it zero), causing the Recursor to check the master in a tight loop.

Lightly tested. It looks like there is a conceptual disconnect between the stored SOA data and `d_refresh` in the Zone objects and I did not fully audit for any other disconnects of that kind.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/rpzloader.cc [hunk 1]
```diff
@@ -373,6 +374,7 @@ void RPZIXFRTracker(const std::vector<ComboAddress>& masters, boost::optional<DN
           refresh = sr->d_st.refresh;
         }
         newZone->setSerial(sr->d_st.serial);
+        newZone->setRefresh(sr->d_st.refresh);
```
[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/rpzloader.cc]: > This overrides a manually set refresh period afaks.

Yes, confirmed in practice.

**Comment by omoerbeek** [comment]
@Habbie can you take  a look at what I pushed a minute ago?

**Comment by Habbie** [comment]
> @Habbie can you take a look at what I pushed a minute ago?

I believe this looks good!

**Comment by omoerbeek** [comment]
Nore that without this PR, a workaround is to explicitly set a refresh time, e.g.
```
pzMaster("194.25.2.170", "test.rpz",{dumpFile="/root/tmp/rpz.dump",seedFile="/root/tmp/rpz.dump", refresh=3600})
```

**Comment by Homas** [comment]
Hey, I'm running free RPZ service (on ioc2rpz) and with PowerDNS 4.3.1 release I observe issues from a few clients if "refresh time" is not set:
- PowerDNS makes 3 or 4 IXFR requests for the same RPZ. It is multiplied by number of feeds (10 in my case). From one client it was pretty brutal - requests were sent every second;
- Looks like SOA refresh time is ignored in this case.
```
Jun  1 15:17:56 host07 pdns_recursor[586]: Getting IXFR deltas for
dns-bh.ioc2rpz from 94.130.x.y:53, our serial: 1590509700
Jun  1 15:17:57 host07 pdns_recursor[586]: Getting IXFR deltas for
dns-bh.ioc2rpz from 94.130.x.y:53, our serial: 1590509700
Jun  1 15:17:58 host07 pdns_recursor[586]: Getting IXFR deltas for
dns-bh.ioc2rpz from 94.130.x.y:53, our serial: 1590509700
Jun  1 15:18:00 host07 pdns_recursor[586]: Getting IXFR deltas for
dns-bh.ioc2rpz from 94.130.x.y:53, our serial: 1590509700
Jun  1 15:18:01 host07 pdns_recursor[586]: Getting IXFR deltas for
dns-bh.ioc2rpz from 94.130.x.y:53, our serial: 1590509700
Jun  1 15:18:02 host07 pdns_recursor[586]: Getting IXFR deltas for
dns-bh.ioc2rpz from 94.130.x.y:53, our serial: 1590509700
Jun  1 15:18:03 host07 pdns_recursor[586]: Getting IXFR deltas for
dns-bh.ioc2rpz from 94.130.x.y:53, our serial: 1590509700
```

**Comment by cmouse** [comment]
you should open new issue.

**Pull request merged**